From 7af9855aa7bbdce6f8ef509d1f181b5d5b0304c5 Mon Sep 17 00:00:00 2001
From: Matthijs van Otterdijk <matthijs@terminusdb.com>
Date: Sat, 13 Aug 2022 13:08:31 +0200
Subject: [PATCH] FIXED: Zero the PL_CRYPTO_HASH_CONTEXT struct on first
 allocation

---
 crypto4pl.c | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/packages/ssl/crypto4pl.c b/packages/ssl/crypto4pl.c
index 0a275a9..17c836f 100644
--- a/packages/ssl/crypto4pl.c
+++ b/packages/ssl/crypto4pl.c
@@ -368,13 +368,8 @@ pl_crypto_hash_context_new(term_t tcontext, term_t options)
   if ( !context )
     return FALSE;
 
+  memset(context, 0, sizeof(*context));
   context->magic    = HASH_CONTEXT_MAGIC;
-  context->ctx      = NULL;
-  context->mac_ctx  = NULL;
-  context->mac_key  = NULL;
-
-  context->parent_stream = NULL;
-  context->hash_stream   = NULL;
 
   if ( !hash_options(options, context) )
     return FALSE;
